spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-configserver/src/test/java/org/springframework/cloud/kubernetes/configserver/KubernetesEnvironmentRepositoryTests.java

```diff
@@ -105,13 +117,14 @@ public static void before() {
 					true);
 			KubernetesClientConfigContext defaultContext = new KubernetesClientConfigContext(coreApi, defaultSource,
 					"default", springEnv);
+			propertySources.add(new KubernetesClientConfigMapPropertySource(defaultContext));
 
-			NormalizedSource devSource = new NamedConfigMapNormalizedSource(applicationName, "dev", false, true);
-			KubernetesClientConfigContext devContext = new KubernetesClientConfigContext(coreApi, devSource, "dev",
+			if ("dev".equals(namespace)) {
```

Comment:
There was a bug here.  These property sources for the dev namespace were being added even if the namespace was not enabled.